src/runtime-rs/crates/hypervisor/src/device/device_manager.rs

```diff
@@ -437,6 +457,59 @@ impl DeviceManager {
 
         Err(anyhow!("ID are exhausted"))
     }
+
+    async fn try_update_device(&mut self, updated_config: &DeviceConfig) -> Result<()> {
+        let device_id = match updated_config {
+            DeviceConfig::ShareFsCfg(config) => {
+                // Try to find the sharefs device. If found, just return.
+                if let Some(target_id) = self.find_device(config.shared_path.clone()).await {
+                    target_id
+                } else {
+                    return Err(anyhow!(
+                        "no matching device was found to do the update operation"
+                    ));
+                }
+            }
+            // TODO for other Device Type
+            _ => {
+                return Err(anyhow!("update device with unsupported device type"));
+            }
+        };
+
+        // get the original device
+        let target_device = self
+            .get_device_info(&device_id)
+            .await
+            .context("get device failed")?;
+
+        // update device with the updated configuration.
+        let updated_device: ArcMutexDevice = match target_device {
+            DeviceType::ShareFs(mut device) => {
+                if let DeviceConfig::ShareFsCfg(config) = updated_config {
+                    // update the mount_config.
+                    device.config.mount_config = config.mount_config.clone();
+                }
+                Arc::new(Mutex::new(device.clone()))
```

Comment:
Is `clone()` necessary here?  `get_device_info()` returns `DeviceType` by value and is commonly implemented by `clone()`ing `self`. So `target_device` is a clone already and owned by this function. If that's the case it might make more sense to make `target_device` `mut` and drop the `clone()` call here. That would leave `target_device` partially moved from but I don't see it referenced after this `match` statement so it should be ok.

Reply:
Fixed